cmd/system-probe/config/adjust.go

```diff
@@ -0,0 +1,125 @@
+		// enable the connections/network check.
+		log.Warn(deprecationMessage(spNS("enabled"), netNS("enabled")))
+		// ensure others can key off of this single config value for NPM status
+		cfg.Set(netNS("enabled"), true)
+	}
+
+	validateString(cfg, spNS("sysprobe_socket"), defaultSystemProbeAddress, ValidateSocketAddress)
+
+	adjustNetwork(cfg)
+	adjustUSM(cfg)
+	adjustSecurity(cfg)
+}
+
+// validateString validates the string configuration value at `key` using a custom provided function `valFn`.
+// If `key` is not set or `valFn` returns an error, the `defaultVal` is used instead.
+func validateString(cfg config.Config, key string, defaultVal string, valFn func(string) error) {
+	if cfg.IsSet(key) {
+		if err := valFn(cfg.GetString(key)); err != nil {
+			log.Errorf("error validating `%s`: %s, using default value of `%s`", key, err, defaultVal)
+			cfg.Set(key, defaultVal)
+		}
+	} else {
+		cfg.Set(key, defaultVal)
+	}
+}
+
+// validateInt validates the int configuration value at `key` using a custom provided function `valFn`.
+// If `key` is not set or `valFn` returns an error, the `defaultVal` is used instead.
+func validateInt(cfg config.Config, key string, defaultVal int, valFn func(int) error) {
+	if cfg.IsSet(key) {
+		if err := valFn(cfg.GetInt(key)); err != nil {
+			log.Errorf("error validating `%s`: %s, using default value of `%d`", key, err, defaultVal)
+			cfg.Set(key, defaultVal)
+		}
+	} else {
+		cfg.Set(key, defaultVal)
+	}
+}
+
+// validateInt64 validates the int64 configuration value at `key` using a custom provided function `valFn`.
+// If `key` is not set or `valFn` returns an error, the `defaultVal` is used instead.
+func validateInt64(cfg config.Config, key string, defaultVal int64, valFn func(int64) error) {
+	if cfg.IsSet(key) {
+		if err := valFn(cfg.GetInt64(key)); err != nil {
+			log.Errorf("error validating `%s`: %s. using default value of `%d`", key, err, defaultVal)
+			cfg.Set(key, defaultVal)
+		}
+	} else {
+		cfg.Set(key, defaultVal)
+	}
+}
+
+// applyDefault sets configuration `key` to `defaultVal` only if not previously set.
+func applyDefault(cfg config.Config, key string, defaultVal interface{}) {
+	if !cfg.IsSet(key) {
+		cfg.Set(key, defaultVal)
+	}
+}
+
+// deprecateString sets `newkey` to the value at `oldkey` and logs a deprecation message,
+// but only if `oldkey` is set and `newkey` is not set.
+func deprecateString(cfg config.Config, oldkey string, newkey string) {
+	deprecateCustom(cfg, oldkey, newkey, func(cfg config.Config) interface{} {
+		return cfg.GetString(oldkey)
+	})
+}
+
+// deprecateString sets `newkey` to the value obtained from `getFn` and logs a deprecation message,
+// but only if `oldkey` is set and `newkey` is not set.
+func deprecateCustom(cfg config.Config, oldkey string, newkey string, getFn func(config.Config) interface{}) {
+	if cfg.IsSet(oldkey) && !cfg.IsSet(newkey) {
```

Comment:
I do think we should log the deprecation warning if `cfg.IsSet(oldkey) = true` (regardless of `cfg.IsSet(newkey)`)